src/sentry/api/authentication.py

```diff
@@ -439,29 +440,17 @@ def authenticate_token(self, request: Request, token_str: str) -> tuple[Any, Any
                         organization.slug != target_org_id_or_slug
                         and organization.id != target_org_id_or_slug
                     ):
-                        # TODO (@athena): We want to raise auth excecption here but to be sure
-                        # I soft launch this by only logging the error for now
-                        # raise AuthenticationFailed("Unauthorized organization access")
-                        logger.info(
-                            "Token has access to organization %s but wants to get access to organization %s: %s",
-                            organization.slug,
-                            target_org_id_or_slug,
-                            request.path_info,
-                        )
+                        raise AuthenticationFailed("Unauthorized organization access.")
                 else:
-                    # TODO (@athena): We want to limit org level token's access to org level endpoints only
-                    # so in the future this will be an auth exception but for now we soft launch by logging an error
-                    logger.info(
-                        "Token has only access to organization %s but is calling an endpoint for multiple organizations: %s",
-                        organization.slug,
-                        request.path_info,
-                    )
+                    # We want to limit org scoped tokens access to org level endpoints only
+                    # Or none org level endpoints that we added special treatments for
+                    if resolved_url.url_name not in ["sentry-api-0-organizations"]:
```

Comment:
It's safe to allowlist usage of this endpoint even though the endpoint is not specific to one organization because this case is already handled in the endpoint code: https://github.com/getsentry/sentry/blob/master/src/sentry/api/endpoints/organization_index.py#L98

So basically if I call this endpoint to list organizations of one person with a token that only has access to one of them the endpoint only returns that one.